### PR TITLE
pass scheduling labels back to rancher

### PR DESCRIPTION
**Pull request by wlan0**
@alena1108

### CODE REVIEW

**Comment by wlan0** [comment]
Addresses https://github.com/rancher/rancher/issues/4849

**Comment by alena1108** [comment]
https://github.com/rancher/rancher/issues/5240

**Comment by qamalbcgdv** [comment]
@alena1108 is there any chance we can also support this?

https://github.com/kubernetes/contrib/tree/master/ingress/controllers/nginx#running-multiple-ingress-controllers

This will give a decent/standard workaround if you want to deploy your own ingress controller. e.g:

```
kubernetes.io/ingress.class: "nginx"
```

If something other than (pick a name, `rancher` for instance) is specified, it won't claim it.

**Comment by alena1108** [comment]
@qamalbcgdv could you file a request for that [here](https://github.com/rancher/rancher/issues)?

File: provider/rancher/rancher.go [hunk 1]
```diff
@@ -462,6 +469,36 @@ func (lbp *LBProvider) getLBServiceForConfig(lbConfigName string) (*client.LoadB
...
+	for k, v1 := range newLabels {
+		if !strings.HasPrefix(k, rancherSchedulerPrefix) {
+			continue
+		}
+		key := rancherSchedulerPrefix + strings.Replace(k[len(rancherSchedulerPrefix):], ".", ":", 1)
+		v2, ok := oldLabels[key]
+		if !ok {
+			return true
+		}
+		if v1 != v2.(string) {
```
Comment (alena1108) [inline provider/rancher/rancher.go]: @wlan0 can you double check if the labels' values are case sensitive in Rancher? If not, do the comparison ignoring the case

File: provider/rancher/rancher.go [hunk 2]
```diff
@@ -517,17 +554,32 @@ func (lbp *LBProvider) createRancherLBService(lbConfig *config.LoadBalancerConfi
...
+	scale := 0
+	if scaleStr, ok := lbConfig.Annotations["scale"]; ok {
+		scale, _ = strconv.Atoi(scaleStr)
+	}
+
+	rancherLabels := map[string]interface{}{}
+
+	for k, v := range lbConfig.Annotations {
+		if strings.HasPrefix(k, rancherSchedulerPrefix) {
+			key := rancherSchedulerPrefix + strings.Replace(k[len(rancherSchedulerPrefix):], ".", ":", 1)
```
Comment (alena1108) [inline provider/rancher/rancher.go]: @wlan0 can we extract this prefix concatenation + replacement to its own method? as the same logic is being called multiple times

Comment (wlan0) [inline provider/rancher/rancher.go]: The two calls have different logic. Notice the second and the third arguments.

**Comment by LoicMahieu** [comment]
@alena1108 @qamalbcgdv I think an issue has been created rancher/rancher#6344

**Pull request merged**